vertical-pod-autoscaler/pkg/recommender/model/types_test.go

```diff
@@ -0,0 +1,166 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"testing"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type ResourcesAsResourceListTestCase struct {
+	name         string
+	resources    Resources
+	humanize     bool
+	resourceList apiv1.ResourceList
+}
+
+func TestResourcesAsResourceList(t *testing.T) {
+	testCases := []ResourcesAsResourceListTestCase{
+		{
+			name: "basic resources without humanize",
+			resources: Resources{
+				ResourceCPU:    1000,
+				ResourceMemory: 1000,
+			},
+			humanize: false,
+			resourceList: apiv1.ResourceList{
+				apiv1.ResourceCPU:    *resource.NewMilliQuantity(1000, resource.DecimalSI),
+				apiv1.ResourceMemory: *resource.NewQuantity(1000, resource.DecimalSI),
+			},
+		},
+		{
+			name: "basic resources with humanize",
+			resources: Resources{
+				ResourceCPU:    1000,
+				ResourceMemory: 262144000, // 250Mi
+			},
+			humanize: true,
+			resourceList: apiv1.ResourceList{
+				apiv1.ResourceCPU:    *resource.NewMilliQuantity(1000, resource.DecimalSI),
+				apiv1.ResourceMemory: resource.MustParse("250.00Mi"),
+			},
+		},
+		{
+			name: "large memory value with humanize",
+			resources: Resources{
+				ResourceCPU:    1000,
+				ResourceMemory: 839500000, // 800.61Mi
+			},
+			humanize: true,
+			resourceList: apiv1.ResourceList{
+				apiv1.ResourceCPU:    *resource.NewMilliQuantity(1000, resource.DecimalSI),
+				apiv1.ResourceMemory: resource.MustParse("800.61Mi"),
+			},
+		},
+		{
+			name: "zero values without humanize",
+			resources: Resources{
+				ResourceCPU:    0,
+				ResourceMemory: 0,
+			},
+			humanize: false,
+			resourceList: apiv1.ResourceList{
+				apiv1.ResourceCPU:    *resource.NewMilliQuantity(0, resource.DecimalSI),
+				apiv1.ResourceMemory: *resource.NewQuantity(0, resource.DecimalSI),
+			},
+		},
+		{
+			name: "large memory value without humanize",
+			resources: Resources{
+				ResourceCPU:    1000,
+				ResourceMemory: 839500000,
+			},
+			humanize: false,
+			resourceList: apiv1.ResourceList{
+				apiv1.ResourceCPU:    *resource.NewMilliQuantity(1000, resource.DecimalSI),
+				apiv1.ResourceMemory: *resource.NewQuantity(839500000, resource.DecimalSI),
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := ResourcesAsResourceList(tc.resources, tc.humanize)
+			if !result[apiv1.ResourceCPU].Equal(tc.resourceList[apiv1.ResourceCPU]) {
+				t.Errorf("expected %v, got %v", tc.resourceList[apiv1.ResourceCPU], result[apiv1.ResourceCPU])
+			}
+			if !result[apiv1.ResourceMemory].Equal(tc.resourceList[apiv1.ResourceMemory]) {
+				t.Errorf("expected %v, got %v", tc.resourceList[apiv1.ResourceMemory], result[apiv1.ResourceMemory])
+			}
+		})
+	}
+}
+
+type HumanizeMemoryQuantityTestCase struct {
+	name   string
+	value  int64
+	wanted string
+}
+
+func TestHumanizeMemoryQuantity(t *testing.T) {
```

Comment:
Would it be possible to add some more "weird" numbers to these tests?
Such as 4992073454

Reply:
of course, Ill add more numbers.